src/js/components/Button/Button.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledButton = styled.button`
+  cursor: pointer;
+  padding: 0.375rem 0.625rem;
+  margin: 0;
+  font-family: inherit;
+  font-size: inherit;
+  border-radius: 0.25rem;
+  font-weight: 500;
+  border: none;
+  display: inline-flex;
+  align-items: center;
+  color: var(--body-text-color);
+  background-color: transparent;
+  transition-property: filter, background, color, opacity;
+  transition-duration: 0.075s;
+  transition-timing-function: ease;
+
+  &:hover {
+    background: var(--body-text-color---opacity-lower);
+  }
+
+  &:active,
+  &:hover:active,
+  &[aria-pressed="true"] {
+    color: var(--body-text-color);
+    background: var(--body-text-color---opacity-lower);
+  }
+
+  &:active,
+  &:hover:active,
+  &:active[aria-pressed="true"] {
+    background: var(--body-text-color---opacity-low);
+  }
+
+  &:disabled,
+  &:disabled:active {
+    color: var(--body-text-color---opacity-low);
+    background: var(--body-text-color---opacity-lower);
+    cursor: default;
+  }
+
+  span {
+    flex: 1 0 auto;
+    text-align: left;
+  }
+
+  kbd {
+    margin-inline-start: 1rem;
+    font-size: 85%;
+  }
+
+  > svg {
+    margin: -0.0835em -0.325rem;
+
+    &:not(:first-child) {
+      margin-left: 0.251em;
+    }
+    &:not(:last-child) {
+      margin-right: 0.251em;
+    }
+  }
+
+  &.is-primary {
+    color: var(--link-color);
+    background: var(--link-color---opacity-lower);
+
+    &:hover {
+      background: var(--link-color---opacity-low);
+    }
+
+    &:active,
+    &:hover:active,
+    &[aria-pressed="true"] {
+      color: white;
+      background: var(--link-color);
+    }
+
+    &:disabled,
+    &:disabled:active {
+      color: var(--body-text-color---opacity-low);
+      background: var(--body-text-color---opacity-lower);
+      cursor: default;
+    }
+  }
+`;
+
+export interface ButtonProps {
+  /**
+   * Is this the principal call to action on the page?
+   */
+  isPrimary?: boolean;
+  /**
+   * Is this the principal call to action on the page?
+   */
+  isPressed?: boolean;
```

Comment:
Can we explicitly define what conventions we use, and eventually enforce them to the extent we can with build tools? There are 4 if I'm not mistaken.

CLJS: `primary?`
CSS: `is-primary`
JS: `isPrimary`
TS: `isPrimary?`

Reply:
On `TS` `isPrimary?` means the prop is optional. For it to have `?` in the prop name it would have to be a string, e.g. `"isPrimary?"`.

I think the only distinction we should have is between camelCase and dash-case. CLJ tools usually convert camelCase to dash-case on interop.

Reply:
I prefer is/has/should prefixes for booleans in TS, since the "?" isn't available to distinguish them.

Reply:
Stuart thinks there's no action here, since it seems we're just following CLJS->JS naming conversion, is there something you'd like to be done here @tangjeff0?

Reply:
Ok